graph/src/util/ethereum.rs

```diff
@@ -123,20 +123,24 @@ pub fn contract_function_with_signature<'a>(
     target_signature: &str,
 ) -> Option<&'a Function> {
     contract.functions().find(|function| {
```

Comment:
Actually this would be better if we checked the mutability in a call to `filter`. Also we should do an exhaustive match on all variants, for better future proofing.